robusta_krr/main.py

```
@@ -147,6 +148,30 @@ def run_strategy(
+                    "node-exporter), 'otel' (OpenTelemetry Collector k8s_cluster and hostmetrics receivers), or "
+                    "'auto' to detect it. Container usage metrics always use cAdvisor names.",
+                    rich_help_panel="Prometheus Settings",
+                ),
+                prometheus_owner_resolution: OwnerResolutionName = typer.Option(
+                    OwnerResolutionName.AUTO.value,
+                    "--prometheus-owner-resolution",
+                    help="How to find the pods of a workload: 'kube-state-metrics' walks the kube_*_owner chain, "
+                    "'recording-rule' uses a single pod owner recording rule, 'auto' prefers the owner chain and "
+                    "falls back to the recording rule.",
+                    rich_help_panel="Prometheus Settings",
+                ),
```

Ruff (0.16.0)

[warning] Do not perform function call `typer.Option` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)